instructions.ts

```diff
@@ -33,30 +32,38 @@ export function action<T>(
     *[Symbol.iterator]() {
       return yield function Action(frame) {
         return shift<Result<T>>(function* (k) {
-          let results = createEventStream<void, Result<T>>();
+          let settle = yield* reset<Resolve<Result<T>>>(function* () {
+            let result = yield* shift<Result<T>>(function* (k) {
+              return k.tail;
```

Comment:
Another shadow k here which might be confusing

Reply:
I think I'm going leave it since `k` is idiomatic. I tried making it:

```ts
 let result = yield* shift<Result<T>>(function* ({tail}) {
   return tail;
});
```
But I don't think it makes it any clearer, because `k` is marker for "Continuation". I also tried extracting it into its own function, but that was weird too. 

Not sure what's best🤷🏻‍♂️